### PR TITLE
[node] add debug options to node bindings

### LINKED ISSUE #3579: Add debug options to node bindings
Add a `debug` option to [`map.render`](https://github.com/mapbox/mapbox-gl-native/tree/master/platform/node#rendering-a-map-tile) with the following sub-options, corresponding to [`MapDebugOptions`](https://github.com/mapbox/mapbox-gl-native/blob/a88e58ef3a90225fcfafca1efab996de6a827b72/include/mbgl/map/mode.hpp#L31-L37):

```
debug: {
  tileBorders: true/false,
  parseStatus: true/false,
  timestamps: true/false,
  collision: true/false
}
```

cc @mikemorris @bsudekum

### PR DESCRIPTION
**Pull request by ansis**
fix #3579 

:eyes: @jfirebaugh @mikemorris

### CODE REVIEW

File: platform/node/src/node_map.cpp [hunk 1]
```diff
@@ -25,6 +25,15 @@ struct NodeMap::RenderOptions {
     unsigned int width = 512;
     unsigned int height = 512;
     std::vector<std::string> classes;
+    bool debugTileBorders = false;
```
Comment (jfirebaugh) [inline platform/node/src/node_map.cpp]: Why not use `mbgl::MapDebugOptions` here?

Comment (ansis) [inline platform/node/src/node_map.cpp]: To `mbgl` stuff out of `NodeMap::ParseOptions`. Should I switch it?

I replaced these three bools with the nested debug struct and forgot to remove them.

Comment (mikemorris) [inline platform/node/src/node_map.cpp]: Pulling in `mbgl::MapDebugOptions` here seems :+1: to me.

Comment (ansis) [inline platform/node/src/node_map.cpp]: switched to using `mbgl::MapDebugOptions` 1934258

File: platform/ios/vendor/SMCalloutView [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 2aede5d8d1577101bf18405246220e7a710df607
+Subproject commit 717078fb6b0d02b944a12f4d40fc15567489d23b
```
Comment (jfirebaugh) [inline platform/ios/vendor/SMCalloutView]: Accidental submodule update?

Comment (ansis) [inline platform/ios/vendor/SMCalloutView]: yeah, thanks for catching that

**Comment by jfirebaugh** [comment]
:+1:

**Pull request merged**